Makefile

```diff
@@ -29,49 +29,67 @@ INCLUDES = -I$(THIRD_PARTY_PATH)/http-parser -I$(MBEDTLS_PATH)/include -I$(ROOT_
 LIBS = $(MBEDTLS_PATH)/library/libmbedx509.a $(MBEDTLS_PATH)/library/libmbedtls.a $(MBEDTLS_PATH)/library/libmbedcrypto.a
 export INCLUDES
 
-UTILS_OBJS = $(UTILS_PATH)/crypto_utils.o $(UTILS_PATH)/serializer.o $(UTILS_PATH)/tryte_byte_conv.o $(UTILS_PATH)/uart_utils.o
+UTILS_OBJS = $(UTILS_PATH)/crypto_utils.o $(UTILS_PATH)/serializer.o $(UTILS_PATH)/tryte_byte_conv.o $(UTILS_PATH)/uart_utils.o $(UTILS_PATH)/utils.o
 # We need to modify this rule here to be compatible to the situation 
 # that we have several different ways of connectivity in the future
 CONNECTIVITY_OBJS = conn_http.o
 
-OBJS = main.o $(HTTP_PARSER_PATH)/http_parser.o $(UTILS_OBJS) $(CONNECTIVITY_OBJS)
+OBJS = main.o http_parser.o $(UTILS_OBJS) $(CONNECTIVITY_OBJS)
+TA_LIB = libta_endpoint.so
 
 .SUFFIXES:.c .o
-.PHONY: all clean test pre-build
+.PHONY: all clean test pre-build help mbedtls_make
 
 all: pre-build ta_client mbedtls_make
 
+## Build ta-client
 ta_client: mbedtls_make $(OBJS)
 	@echo Linking: $@ ....
 	$(CC) -o $@ $(OBJS) $(LIBS)
 
 mbedtls_make:
-	@for dir in $(MBEDTLS_PATH); do \
-		$(MAKE) -C $$dir ; \
-		if [ $$? != 0 ]; then exit 1; fi; \
-	done
+	$(MAKE) -C $(MBEDTLS_PATH) lib
+
+$(TA_LIB): mbedtls_make $(UTILS_OBJS) $(CONNECTIVITY_OBJS) http_parser.o
+	$(CC) -shared -o $@ $(UTILS_OBJS) $(CONNECTIVITY_OBJS) http_parser.o $(LIBS)
 
 conn_http.o: connectivity/conn_http.c
 	@echo Compiling $@ ...
-	$(CC) -v -c $(CFLAGS) $(INCLUDES) -o $@ $<
-
+	$(CC) -c $(CFLAGS) $(INCLUDES) -o $@ $<
+http_parser.o: $(THIRD_PARTY_PATH)/http-parser/http_parser.c
+	$(CC) -c $(CFLAGS) $(INCLUDES) -o $@ $<
 main.o: main.c $(UTILS_OBJS)
 	@echo Compiling: $< ....
 	$(CC) -c $(CFLAGS) $(INCLUDES) -o $@ $<
 $(UTILS_OBJS):
 	$(MAKE) -C $(UTILS_PATH)
+
+## Build tests
 test: $(TEST_PATH) $(UTILS_OBJS)
 	$(MAKE) -C $(TEST_PATH)
 
-clean: clean_client clean_third_party clean_test
+## Build ta-endpoint legato app
+ta_legato_app:
+	# TODO: notify should make ta_lib with arm compiler first
+	mkapp -v -t wp77xx ta-endpoint.adef
```

Comment:
You shall detect Legato AF in advance.

Reply:
I had try this. But how can we detect current bash setting like after command `source legat_bashrc` environment inside makefile?

Reply:
It sounds good. But, you shall improve the build instructions.